src/index.ts

```diff
@@ -120,18 +136,19 @@ export class WalletPluginMetaMask extends AbstractWalletPlugin implements Wallet
         return data
     }
 
-    async initialize() {
+    async initialize(context?: LoginContext) {
         if (!this.provider) {
             this.provider = await getSnapsProvider()
         }
         if (this.provider && !this.installedSnap) {
             this.isFlask = await checkIsFlask(this.provider)
             await this.setSnap()
             if (!this.installedSnap) {
-                await this.requestSnap()
-                if (this.installedSnap) {
-                    await this.initialize()
-                }
+                context?.ui?.prompt({
+                    title: 'Snap not found',
+                    body: `Plese visit ${this.setupPageUrl} for help setting up the snap.`,
```

Comment:
Spelling - and should this be text in the body text, or an actual clickable link? AFAIK right now this is just going to put the text in place and users are going to have to copy/paste it?

Reply:
We also need to update this text, it needs to be more user friendly. I have no suggestions at the moment but figured I'd just mention it.

Reply:
lol I had to read it twice to find the spelling error 😂